docs/source/peers/peers.md

```diff
@@ -1,154 +1,183 @@
+3. **Endorsement delegation** --- If A1 has delegated transaction endorsement for a ledger update, the gateway service on P1: a) requests endorsement from a peer in its own organization (selecting the available peer with the highest block height, which can be P1); b) reads the set of endorsement policies for the updated keys; and c) forwards the proposal to endorsement peers in the organizations required to endorse this transaction.
+4. **Endorsement collection** --- P1 collects the transaction endorsement (or query) results from each organization and returns the proposal response to the client application A1. For ledger queries, the process is now complete. For ledger updates, if the proposal is endorsed, P1 forwards the transaction to the channel ordering service.
+5. **Endorsement retries** --- If the transaction proposal is *not* endorsed, the gateway service will attempt retries on peers that would satisfy endorsement for a ledger update (unless private data has been submitted). If the proposal is *not* endorsed due to detection of transient data --- which often contains private data --- P1 returns a request to the client (A1) for explicit specification of the organizations that are members of the private data collection(s). Once A1 is updated and resubmitted in consideration of private data, the gateway repeats the process by requesting endorsement from the specified organizations and collecting the results.
+6. **Endorsement failure** --- If the returned endorsements do not satisfy the endorsement policy requirements, P1 returns a failure message to the client and P1 updates its ledger with *only* a record of the failed transaction proposal. For a failed endorsement, no other ledger or state database update occurs on any peer, and the transaction workflow is now complete.
+7. **Endorsement success** --- If the returned endorsements satisfy the endorsement policy requirements, P1 submits the transaction to the ordering service. Once the ordering service validates the transaction, P1 returns a transaction envelope with the endorsements to the client for signing.
+8. **Transaction signing** --- The client (A1) returns the signed transaction to P1, which forwards the transaction to O1 for commitment to the ledger (along with other transactions which are first ordered and packaged together in blocks for the ledger update.)
+9. **Ledger commitment** --- O1 collects transactions from across the channel and packages them into ordered blocks and distributes the blocks to all peers, including P1. P1 (and all other channel peers) validates the transaction before committing it to the ledger (L1). The commit is an immutable update written to the channel ledger, and the world state (the effect of the transaction, such as a change in asset ownership) is also updated on each peer.
+10. **Commit event** --- A1 receives a commit status event from P1 with proof of the ledger commit.
+
+### Querying the Ledger
```

Comment:
I find it valuable to have an e2e in one place, rather than splitting it up into phases and subsequent details in each phase, in different doc sections. This is arguable however. ... Some redundancy can be a good thing for readers, will re-visit.

Reply:
returning to the 3 phases.

Reply:
Maybe should keep the legacy 3 phase sections and just add the clearer steps to it for gateway.